### PR TITLE
[Feature] add user spec in instance

### PR DESCRIPTION
**Pull request by Sispheor**

### CODE REVIEW

File: service_catalog/api/serializers/instance_serializer.py [hunk 1]
```diff
@@ -15,6 +15,12 @@ class Meta:
         fields = '__all__'
 
 
+class RestrictedInstanceReadSerializer(InstanceReadSerializer):
+    class Meta:
+        model = Instance
+        exclude = ["spec"]
+
+
 class InstanceSerializer(serializers.ModelSerializer):
```
Comment (EliasBoulharts) [inline service_catalog/api/serializers/instance_serializer.py:24]: In request we use AdminRequestSerializer and RequestSerializer

File: templates/service_catalog/common/instance-details.html [hunk 2]
```diff
@@ -102,6 +103,9 @@ <h3 class="card-title">Details</h3>
                             <div class="tab-pane" id="spec">
                                 <pre><code class="language-json">{{ instance.spec | pretty_json }}</code></pre>
                             </div>
+                            <div class="tab-pane" id="user-spec">
```
Comment (EliasBoulharts) [inline templates/service_catalog/common/instance-details.html:106]: if needed to hide in HTML generated by user

File: tests/test_service_catalog/test_api/test_instance/test_instance_retrieve.py [hunk 3]
```diff
@@ -10,19 +10,24 @@ def setUp(self):
...
         self.assertTrue("id" in response.json())
         self.assertTrue("name" in response.json())
-        self.assertTrue("spec" in response.json())
         self.assertTrue("state" in response.json())
         self.assertTrue("billing_group" in response.json())
         self.assertTrue("service" in response.json())
-        self.assertTrue("spec" in response.json())
+        self.assertTrue("user_spec" in response.json())
+        if is_admin:
+            self.assertTrue("spec" in response.json())
```
Comment (EliasBoulharts) [inline tests/test_service_catalog/test_api/test_instance/test_instance_retrieve.py:22]: assertIn

File: tests/test_service_catalog/test_api/test_instance/test_instance_retrieve.py [hunk 4]
```diff
@@ -10,19 +10,24 @@ def setUp(self):
...
+        else:
+            self.assertFalse("spec" in response.json())
```
Comment (EliasBoulharts) [inline tests/test_service_catalog/test_api/test_instance/test_instance_retrieve.py:24]: assertNotIn

**Review by EliasBoulharts — CHANGES_REQUESTED** [review]
some small changes to do

File: templates/service_catalog/common/instance-details.html (hunk 2 above)
Comment (Sispheor) [reply templates/service_catalog/common/instance-details.html:108]: done

**Pull request merged**